mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -286,6 +294,41 @@ class LogisticRegressionModel private[ml] (
 
   override val numClasses: Int = 2
 
+  private var trainingSummary: Option[BinaryLogisticRegressionTrainingSummary] = None
+
+  /**
+   * Gets summary of model on training set. An exception is
+   * thrown if `trainingSummary == None`.
+   */
+  def summary: BinaryLogisticRegressionTrainingSummary = trainingSummary match {
```

Comment:
This and other methods within LogisticRegressionModel should return LogisticRegressionTrainingSummary or LogisticRegressionSummary, not specific binary/multiclass versions.